### PR TITLE
Replace the HTTP Client from Apache httpclient5 to java.net.http

### PR DESCRIPTION
**Pull request by 41zu**
+ upgraded dependencies and plugins
+ added SteamApiKeyException for when the API Key is null

### CODE REVIEW

**Comment by lpradel** [comment]
Hi thanks a lot for your latest contribution! I will try to review ASAP. Sorry for the inconvenience but could you re-integrate with master? I merged Joey's changes as he was first as 1.6.0 so you should make this into 1.6.1.

Thanks!

**Comment by lpradel** [comment]
Actually on second thought I think this warrants 1.7.0 !

**Comment by 41zu** [comment]
I rebased my fork with master,

and I also think 1.7.0 ist required.

**Comment by 41zu** [comment]
@lpradel do you have any suggestions for improvement or objections for the code in this MR?

File: README.md [hunk 1]
```diff
@@ -193,6 +192,7 @@ mvn clean deploy -P release
 
 ## History
 
+- Version 1.7.0: Replace the internally used HTTP Client from Apache httpclient5 to java.net.http (needs Java 11)
```
Comment (lpradel) [inline README.md:195]: For completeness of the commit/merge you could also set the actual `<version>` in `pom.xml` to 1.7.0 😅

File: pom.xml [hunk 2]
```diff
@@ -11,32 +11,30 @@
 
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <java.version.prop>1.8</java.version.prop>
+        <java.version.prop>11</java.version.prop>
```
Comment (lpradel) [inline pom.xml:14]: Just a minor nitpick but I would prefer if you would drop the `.prop` suffix it's used no where else and imo adds no value

File: src/main/java/com/lukaspradel/steamapi/webapi/request/SteamWebApiRequestHandler.java [hunk 3]
```diff
@@ -35,45 +33,51 @@ URI getRequestUrl(SteamWebApiRequest request) throws SteamApiException {
...
+		)
+		.map(Object::toString)
+		.collect(joining("/", "/", ""));
 	}
 
-	List<NameValuePair> getRequestParameters(Map<String, String> parametersMap) {
-
-		List<NameValuePair> nvps = new ArrayList<>();
+	String getRequestQuery(Map<String, String> parameters) throws SteamApiException {
+		if (getKey() == null)
```
Comment (lpradel) [inline src/main/java/com/lukaspradel/steamapi/webapi/request/SteamWebApiRequestHandler.java:53]: Again just a minor nitpick but for uniformity/consistency I would prefer if you used braces even for single-line `if` / `for`

File: src/main/java/com/lukaspradel/steamapi/webapi/request/SteamWebApiRequestHandler.java [hunk 4]
```diff
@@ -35,45 +33,51 @@ URI getRequestUrl(SteamWebApiRequest request) throws SteamApiException {
...
+			throw new SteamApiKeyException("Steam API key is not present or null");
+
+		for (var e : parameters.entrySet())
```
Comment (lpradel) [inline src/main/java/com/lukaspradel/steamapi/webapi/request/SteamWebApiRequestHandler.java:56]: see above

File: src/main/java/com/lukaspradel/steamapi/webapi/request/SteamWebApiRequestHandler.java [hunk 5]
```diff
@@ -82,33 +86,26 @@ URI getRequestUri(String scheme, String host, String path, List<NameValuePair> p
...
-			}
-			return getHttpResponseAsString(response);
-		} catch (IOException | ParseException e) {
+		HttpRequest getRequest = HttpRequest.newBuilder(requestUrl).build();
+
+		try {
+			var response = client.send(getRequest, BodyHandlers.ofString());
+			int statusCode = response.statusCode();
+
+			if (Integer.toString(statusCode).startsWith("20"))
```
Comment (lpradel) [inline src/main/java/com/lukaspradel/steamapi/webapi/request/SteamWebApiRequestHandler.java:95]: Compiler will likely optimize this anyway but for clarity I would prefer to change this into
```
if () {
} else if {
} else {
}
```

if it's ok with you

File: src/main/java/com/lukaspradel/steamapi/webapi/request/SteamWebApiRequestHandler.java [hunk 6]
```diff
@@ -35,45 +33,51 @@ URI getRequestUrl(SteamWebApiRequest request) throws SteamApiException {
...
-	URI getRequestUri(String scheme, String host, String path, List<NameValuePair> parameters) throws SteamApiException {
 		try {
-			return new URIBuilder()
-					.setScheme(scheme)
-					.setHost(host)
-					.setPath(path)
-					.setParameters(parameters)
-					.build();
+			return new URI(uri.toString());
 		} catch (URISyntaxException e) {
```
Comment (lpradel) [inline src/main/java/com/lukaspradel/steamapi/webapi/request/SteamWebApiRequestHandler.java:81]: Honestly not sure if it was even covered before, but in my IDE coverage it says that this `catch` case is not covered by the unit test. Maybe you could add one for completeness sake?

**Comment by lpradel** [comment]
Sorry for the delay, was away on leave in the pyrenees 🏍️

**Comment by 41zu** [comment]
Thank you for your feedback, I will try to address your suggestions

**Comment by lpradel** [comment]
I went ahead and made the suggested changes myself, hope it's ok with you

**Comment by lpradel** [comment]
Thanks again for your contribution!

**Comment by 41zu** [comment]
Thank you for taking the time and making the effort to implement the changes. Unfortunately, I haven't had much time lately.

**Pull request merged**